File: android/src/main/kotlin/space/variance/web3_signers/PlatformAuthenticatorErrors.kt
package space.variance.web3_signers

val KEY_ALREADY_EXISTS = FlutterError("KEY_ALREADY_EXISTS", "Key already exists", null)

val KEY_NOT_FOUND = FlutterError("KEY_NOT_FOUND", "Key not found", null)

val BIOMETRICS_NOT_AVAILABLE = FlutterError("BIOMETRICS_NOT_AVAILABLE", "Biometrics not available", null)

val NO_ACTIVITY = FlutterError("NO_ACTIVITY", "No foreground activity available yet", null)

fun KEY_GENERATION_FAILED(details: String? = "Failed to generate key pair"): FlutterError {
    return FlutterError("KEY_GENERATION_FAILED", details, null)
}

fun PUBLIC_KEY_RETRIEVAL_FAILED(details: String? = "Failed to retrieve public key"): FlutterError {
    return FlutterError("PUBLIC_KEY_RETRIEVAL_FAILED", details, null)
}

fun SIGNING_FAILED(details: String? = "Failed to sign data"): FlutterError {
    return FlutterError("SIGNING_FAILED", details, null)
}

fun PLATFORM_ERROR(details: String? = "Platform error"): FlutterError {
    return FlutterError("PLATFORM_ERROR", details, null)
}

fun USER_CANCELED(details: String? = "User canceled authentication"): FlutterError {
    return FlutterError("USER_CANCELED", details, null)
}

fun LOCKED_OUT(details: String? = "Biometrics locked out"): FlutterError {
    return FlutterError("LOCKED_OUT", details, null)
}

fun AUTH_ERROR(details: String? = "Authentication failed"): FlutterError {
    return FlutterError("AUTH_ERROR", details, null)
}

File: darwin/web3_signers/Sources/web3_signers/PlatformAuthenticatorErrors.swift
let KEY_ALREADY_EXISTS = PigeonError(
    code: "KEY_ALREADY_EXISTS", message: "Key already exists", details: nil)

let KEY_NOT_FOUND = PigeonError(code: "KEY_NOT_FOUND", message: "Key not found", details: nil)

let PUBLIC_KEY_RETRIEVAL_FAILED = PigeonError(
    code: "PUBLIC_KEY_RETRIEVAL_FAILED", message: "Failed to retrieve public key", details: nil)

let BIOMETRICS_NOT_AVAILABLE = PigeonError(
    code: "BIOMETRICS_NOT_AVAILABLE", message: "Biometrics not available", details: nil)

let USER_CANCELED = PigeonError(
    code: "USER_CANCELED", message: "User canceled authentication", details: nil)

let LOCKED_OUT = PigeonError(code: "LOCKED_OUT", message: "Biometrics locked out", details: nil)

let AUTH_ERROR = PigeonError(code: "AUTH_ERROR", message: "Authentication failed", details: nil)

func KEY_GENERATION_FAILED(details: String = "Failed to generate key pair") -> PigeonError {
    return PigeonError(code: "KEY_GENERATION_FAILED", message: details, details: nil)
}

func SIGNING_FAILED(details: String = "Failed to sign data") -> PigeonError {
    return PigeonError(code: "SIGNING_FAILED", message: details, details: nil)
}

func PLATFORM_ERROR(details: String = "Platform error") -> PigeonError {
    return PigeonError(code: "PLATFORM_ERROR", message: details, details: nil)
}
